akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MiscDirectivesSpec.scala

```diff
@@ -100,6 +100,28 @@ class MiscDirectivesSpec extends RoutingSpec {
       }
     }
 
+    "apply if form data is fully consumed into a map" in {
+      val route =
+        withSizeLimit(64) {
+          formFieldMap { _ ⇒
+            completeOk
+          }
+        }
+
+      Post("/abc", formDataOfSize(32)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+
+      Post("/abc", formDataOfSize(128)) ~> Route.seal(route) ~> check {
+        status shouldEqual StatusCodes.BadRequest
+        responseAs[String] shouldEqual "The request content was malformed:\n" +
+          "EntityStreamSizeException: actual entity size (Some(134)) " +
+          "exceeded content length limit (64 bytes)! " +
+          "You can configure this by setting `akka.http.[server|client].parsing.max-content-length` " +
+          "or calling `HttpEntity.withSizeLimit` before materializing the dataBytes stream."
```

Comment:
Good to have this test here, so we don't accidentally expose sensitive information by including the exception message in the response. I don't think we need to consider the content length limit sensitive, as it can be found by trial and error anyway.